### DISCUSSION TITLE
Filters enhancement

### DISCUSSION THREAD

**Discussion by maurizio-talpo** [WinMerge/winmerge#1128] [Ideas]
It would be very useful to be able to configure filters with 4 entries:
- included files list
- excluded files list
- included folders list
- excluded folders list

**Comment by Cromagnon31** [comment]
Yes, it's not possible to make easy filtering with only exclusive include or exclude modes.
For example I need to compare .asm source files but not those generated by compiler in Debug or Release directory.
But I need to include binary files who are also in Debug and Release directory. Then I need both Include and Exclude.
Could we expect having such features ?

But it could be fine too if it was possible not diplay empty trees due to masking of ignored / only at left / only at right files.
Visibility will be greately increased.

**Comment by maurizio-talpo** [comment]
Yes, you hit the nail on the head.

We are a software house. Compatibly with our free-time, we could 
contribute to the development.
If you are interested, to be quicker, we need help to understand in 
which part of the code the filters are managed.

Thank you.


TALPO Maurizio
Tel. +39-392-9898174
--------------------
AT Automation srl
Partita IVA 11326150015
Registro imprese di Torino, REA TO-1204473
Codice Univoco SDI A4707H7
Sede legale: Via Cuorgnè, 53/b - 10098 Rivoli (TO) - Italia
Sede operativa: Via Collegno, 116 - 10091 Alpignano (TO) - Italia
Telefono +39-011-0448365
Fax +39-011-0441052
PEC ***@***.***
--------------------------------------
Gli eventuali allegati a questo messaggio sono stati creati con:
Any attachments to this message have been created with:
http://www.7-zip.org/
http://www.libreoffice.org/
http://www.pdfforge.org/

**Comment by sdottaka** [comment]
Implemented by PR #1179
